apps/accounts/api/automations/change_secret_dashboard.py

```diff
@@ -146,24 +148,33 @@ def get(self, request, *args, **kwargs):
             })
 
         if _all or query_params.get('total_count_ongoing_change_secret'):
-            execution_ids = []
-            inspect = app.control.inspect()
-            active_tasks = inspect.active()
-            if active_tasks:
-                for tasks in active_tasks.values():
-                    for task in tasks:
-                        _id = task.get('id')
-                        name = task.get('name')
-                        tp = task.get('kwargs', {}).get('tp')
-                        if name == self.task_name and tp == self.tp:
-                            execution_ids.append(_id)
-
-            snapshots = AutomationExecution.objects.filter(id__in=execution_ids).values_list('snapshot', flat=True)
-
-            asset_ids = {asset for i in snapshots for asset in i.get('assets', [])}
-            account_ids = {account for i in snapshots for account in i.get('accounts', [])}
-            data['total_count_ongoing_change_secret'] = len(execution_ids)
-            data['total_count_ongoing_change_secret_assets'] = len(asset_ids)
-            data['total_count_ongoing_change_secret_accounts'] = len(account_ids)
+            ongoing_counts = cache.get(self.ongoing_change_secret_cache_key)
+            if ongoing_counts is None:
+                execution_ids = []
+                inspect = app.control.inspect()
+                active_tasks = inspect.active()
+                if active_tasks:
+                    for tasks in active_tasks.values():
+                        for task in tasks:
+                            _id = task.get('id')
+                            name = task.get('name')
+                            tp = task.get('kwargs', {}).get('tp')
+                            if name == self.task_name and tp == self.tp:
+                                execution_ids.append(_id)
+
+                snapshots = AutomationExecution.objects.filter(id__in=execution_ids).values_list('snapshot', flat=True)
+
+                asset_ids = {asset for i in snapshots for asset in i.get('assets', [])}
+                account_ids = {account for i in snapshots for account in i.get('accounts', [])}
+
+                ongoing_counts = (len(execution_ids), len(asset_ids), len(account_ids))
+                data['total_count_ongoing_change_secret'] = ongoing_counts[0]
+                data['total_count_ongoing_change_secret_assets'] = ongoing_counts[1]
+                data['total_count_ongoing_change_secret_accounts'] = ongoing_counts[2]
+                cache.set(self.ongoing_change_secret_cache_key, ongoing_counts, 60)
+            else:
+                data['total_count_ongoing_change_secret'] = ongoing_counts[0]
+                data['total_count_ongoing_change_secret_assets'] = ongoing_counts[1]
+                data['total_count_ongoing_change_secret_accounts'] = ongoing_counts[2]
 
         return JsonResponse(data, status=200)
```

Comment:
I suggest the following improvements:
1) Remove unnecessary blank lines at the end of each function to make them more readable.
2) Replace "tp" with a variable instead of hardcoding it in one single call, which makes the code cleaner and easier to read.
3) Use underscores (_) as the parameter names instead of periods (.) in some cases where they could be misinterpreted.
4) Add meaningful comments within functions that describe their purpose or actions taken within them for better documentation.

Note: The specific implementation details will depend heavily on the actual requirements defined elsewhere like project structure and business logic.